### PR TITLE
Hindsight

### PR DESCRIPTION
**Pull request by hed-fb**
Adding hindsight to turbinia. Fixed merge issues in PR #393

### CODE REVIEW

File: turbinia/turbiniactl.py [hunk 1]
```diff
@@ -403,6 +417,18 @@ def main():
         name=args.name, disk_name=args.disk_name,
         embedded_path=args.embedded_path, mount_partition=args.mount_partition,
         project=args.project, zone=args.zone, source=args.source)
+  elif args.command == 'hindsight':
+    if args.format not in ['xlsx', 'sqlite', 'jsonl']:
+      log.error('Invalid output format.')
+      sys.exit(1)
+    if args.browser_type not in ['Chrome', 'Brave']:
```
[the remark this answers is not in the document]
Comment (hed-fb) [reply turbinia/turbiniactl.py]: It is actually case sensitive.

File: turbinia/evidence.py [hunk 2]
```diff
@@ -199,6 +199,23 @@ class Directory(Evidence):
   pass
 
 
+class ChromiumProfile(Evidence):
+  """Chromium based browser profile evidence.
+
+  Attributes:
+    browser_type: The type of browser.
+      Supported options are Chrome (default) and Brave.
+    format: Output format (default is XLSX, other option is SQLite)
```
[the remark this answers is not in the document]
Comment (hed-fb) [reply turbinia/evidence.py]: Added jsonl and changed the default to sqlite.

File: turbinia/workers/hindsight.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    super(HindsightTask, self).__init__()
+
+  def run(self, evidence, result):
+    """Task to execute hindsight."""
+
+    # Create the new Evidence object that will be generated by this Task.
+    output_evidence = TextFile()
+
+    # Create a path that we can write the new file to.
+    output_file_path = os.path.join(self.output_dir, '{0:s}'.format(self.id))
```
[the remark this answers is not in the document]
Comment (hed-fb) [reply turbinia/workers/hindsight.py]: Done.

**Comment by aarontp** [comment]
Thanks @hed-fb - I see you've made some changes to update some things here.  Please ping me in a comment when you're ready for re-review.  Thanks!

**Comment by hed-fb** [comment]
Thanks, Aaron! I updated the code based on your input with a few exceptions due to case sensitivity. It should be good to review now.

File: turbinia/workers/hindsight.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    super(HindsightTask, self).__init__()
+
+  def run(self, evidence, result):
+    """Task to execute hindsight."""
+
+    # Create the new Evidence object that will be generated by this Task.
+    output_evidence = TextFile()
+
+    # Create a path that we can write the new file to.
+    output_file_path = os.path.join(self.output_dir, 'hindsight_report')
```
Comment (aarontp) [inline turbinia/workers/hindsight.py]: Just one small thing here:  Can we set the output file extension to `evidence.output_format`?

Comment (hed-fb) [reply turbinia/workers/hindsight.py]: Sorry I saw your comment earlier but hindsight accepts the output path without the extension and adding that can cause double extensions.

Comment (aarontp) [reply turbinia/workers/hindsight.py]: Ah, I see.  Ok, thanks!  LG then, I'll merge now.

**Pull request merged**